sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs

```diff
@@ -82,22 +84,59 @@ internal async ValueTask<int> AddBatchActivityAsync(Batch<Activity> batchActivit
         private static TelemetryItem GeneratePartAEnvelope(Activity activity)
         {
             TelemetryItem telemetryItem = new TelemetryItem(PartA_Name_Mapping[activity.GetTelemetryType()], activity.StartTimeUtc);
-            // TODO: Validate if Azure SDK has common function to generate role instance
-            telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()] = "testRoleInstance";
-
+            GetRoleInfo(activity.GetResource(), out var roleName, out var roleInstance);
+            telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()] = roleName;
+            telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()] = roleInstance;
             telemetryItem.Tags[ContextTagKeys.AiOperationId.ToString()] = activity.TraceId.ToHexString();
             if (activity.Parent != null)
             {
                 telemetryItem.Tags[ContextTagKeys.AiOperationParentId.ToString()] = activity.Parent.SpanId.ToHexString();
             }
-
-            // TODO: "ai.location.ip"
             // TODO: Handle exception
             telemetryItem.Tags[ContextTagKeys.AiInternalSdkVersion.ToString()] = SdkVersionUtils.SdkVersion;
 
             return telemetryItem;
         }
 
+        internal static void GetRoleInfo(Resource resource, out string roleName, out string roleInstance)
```

Comment:
minor - in general, if a method is returning void, then it shouldn't be named "Get..". I think ExtractRoleInfo, or PopulatRoleInfoFromResource are better.